src/hakunapi-jsonfg/src/main/java/fi/nls/hakunapi/jsonfg/JSONFG.java

```diff
@@ -0,0 +1,132 @@
+    public static final byte[] INTERVAL = "interval".getBytes();
+
+    public static final byte[] GEOMETRY = "geometry".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PLACE = "place".getBytes();
+
+    public static final byte[][] CONF = new byte[][] {
+            //
+            "http://www.opengis.net/spec/json-fg-1/0.2/conf/core".getBytes(),
+            // "http://www.opengis.net/spec/json-fg-1/0.2/conf/types-schemas".getBytes()
+    };
+
+    /*
+     * TODO: JSONFG schema differs from JSON schema. We'll need JSONFG schema
+     * resource or SchemaFormatFactorySpi or such to support providing JSONFG schema
+     */
+    public static String getJSONFGSchema(FeatureType ft) {
+        return null;
+    }
+
+    public static Integer getGeometryDimension(HakunaPropertyGeometry geomType) {
+        if (geomType == null) {
+            return null;
+        }
+
+        Integer geometryDimension = null;
+
+        switch (geomType.getGeometryType()) {
+        case POINT:
+        case MULTIPOINT:
+            geometryDimension = 0;
+            break;
+        case LINESTRING:
+        case MULTILINESTRING:
+            geometryDimension = 1;
+            break;
+        case POLYGON:
+        case MULTIPOLYGON:
+            geometryDimension = 2;
+            break;
+        case GEOMETRY:
+        case GEOMETRYCOLLECTION:
+        default:
+            break;
+
+        }
+        return geometryDimension;
+    }
+
+    public static void writePlace(HakunaJsonWriter json, HakunaGeometry placeGeometry, HakunaGeometry geometry,
+            JSONFGGeometryWriter placeJson, GeometryWriter geometryJson, ProjectionTransformer outputCrs84Proj)
+            throws Exception {
+        if (placeGeometry != null) {
+
+            placeGeometry.write(placeJson);
+            if (geometry != null) {
+                geometry.write(geometryJson);
+            }
+        } else if (geometry != null) {
+            geometry.write(geometryJson);
+        } else {
+            json.writeFieldName(GEOMETRY);
+            json.writeNull();
+        }
+    }
+
+    public static void writeTemporal(HakunaJsonWriter json, LocalDate date, Instant timestamp) throws IOException {
+        if (date == null && timestamp == null) {
+            return;
+        }
+
+        json.writeFieldName(JSONFG.TIME);
+        json.writeStartObject();
+
+        if (timestamp != null) {
+            json.writeFieldName(JSONFG.TIMESTAMP);
+            json.writeStringUnsafe(timestamp.toString());
+        } else if (date != null) {
+            json.writeFieldName(DATE);
+            json.writeStringUnsafe(date.toString());
+        }
+
+        json.writeEndObject();
+
+    }
+
+    public static HakunaGeometry getFootprintGeometry(HakunaGeometry placeGeometry,
+            ProjectionTransformer outputCrs84Proj) {
+
+        Geometry src = placeGeometry.toJTSGeometry();
+
+        Geometry dup = src.copy();
```

Comment:
Looking at this I should probably add heavy comments that informs others that `HakunaGeometry.toJTSGeometry()` and `ProjectionHelper.reproject()` are allowed to return `Geometry`s that might be backed by the same mutable data.